packages/ui/src/ui/components/JobStatisticTable/JobStatisticTable.tsx

```diff
@@ -0,0 +1,180 @@
+}
+
+export function JobStatisticTableStaticCell({item, aggregation}: StatisticProps) {
+    if (item.isLeafNode) {
+        const statistic: LeafStatistic = item.attributes.value as LeafStatistic;
+
+        if (aggregation === 'avg') {
+            return <Avg item={item} />;
+        } else {
+            return hammer.format['Number'](statistic[aggregation]);
+        }
+    }
+
+    return hammer.format.NO_VALUE;
+}
+
+interface ItemState {
+    empty: boolean;
+    collapsed: boolean;
+    visible: boolean;
+}
+
+interface MetricProps<Item extends Partial<JobTreeItem>> {
+    item: Item;
+    itemState: ItemState;
+    toggleItemState: Function;
+    renderValue: (item: Item) => React.ReactChild;
+    minWidth?: number;
+}
+
+export function ExpandedCell<Item extends Partial<JobTreeItem>>({
+    item,
+    itemState,
+    toggleItemState,
+    minWidth = undefined,
+    renderValue,
+}: MetricProps<Item>) {
+    const offsetStyle = React.useMemo(() => {
+        return {minWidth, paddingLeft: (item?.level || 0) * LEVEL_OFFSET};
+    }, [item.level, minWidth]);
+
+    const toggleItemAndTreeState = useCallback(() => {
+        if (!itemState.empty) {
+            toggleItemState();
+        }
+    }, [itemState, toggleItemState]);
+
+    if (item.isLeafNode) {
+        return (
+            <span className={block('metric')} style={offsetStyle}>
+                <Icon awesome="chart-line" className={block('metric-icon')} />
+
+                <span>{renderValue(item)}</span>
+            </span>
+        );
+    } else {
+        return (
+            <span className={block('group')} style={offsetStyle} onClick={toggleItemAndTreeState}>
+                <Icon
+                    awesome={itemState.collapsed || itemState.empty ? 'angle-down' : 'angle-up'}
+                    className={block('group-icon-toggler')}
+                />
+                <Icon
+                    awesome={itemState.collapsed || itemState.empty ? 'folder' : 'folder-open'}
+                    className={block('group-icon')}
+                />
+                <span>{renderValue(item)}</span>
+            </span>
+        );
+    }
+}
+
+export type TableProps<Item extends Partial<JobTreeItem>> = {
+    columns?: {
+        mode: string;
+        sets?: {
+            default?: {
+                items: string[];
+            };
+        };
+        items: Record<
+            string,
+            {
+                sort: boolean;
+                align: string;
+            }
+        >;
+    };
+    theme: string;
+    size: string;
+    striped: boolean;
+    tree: boolean;
+    computeKey(item: Item): string;
+};
+
+export type JobStatisticTableTemplate<Item extends Partial<JobTreeItem>> = {
+    [name: string]: (
+        item: Item,
+        colName: string,
+        toggleItemState: Function,
+        itemState: ItemState,
+    ) => React.ReactChild | null;
+};
+
+export function GeneralJobStatisticTable<Item extends Partial<JobTreeItem>>({
```

Comment:
We don't need `Job` world in all named entities in the file, now it is just StatisticsTable